mxcube3/core/components/samplechanger.py

```diff
@@ -432,53 +433,66 @@ def queue_mount_sample(view, data_model, centring_done_cb, async_result):
         signals.loaded_sample_changed(sample_mount_device.get_loaded_sample())
         logging.getLogger("user_level_log").info("Sample loaded")
         dm = HWR.beamline.diffractometer
-        if dm is not None:
-            try:
-                dm.connect("centringAccepted", centring_done_cb)
-                centring_method = queue_entry.CENTRING_METHOD
-
-                if centring_method == queue_entry.CENTRING_METHOD.MANUAL:
-                    msg = "Manual centring used, waiting for" + " user to center sample"
-                    log.warning(msg)
-                    dm.start_centring_method(dm.MANUAL3CLICK_MODE)
-                elif centring_method in [
-                    queue_entry.CENTRING_METHOD.LOOP,
-                    queue_entry.CENTRING_METHOD.FULLY_AUTOMATIC,
-                ]:
-                    if not dm.current_centring_procedure:
-                        dm.start_centring_method(dm.C3D_MODE)
-
-                    # NBNB  BUG . self and app are not avialble here
-                    if mxcube.AUTO_MOUNT_SAMPLE:
-                        msg = "Going to save centring automatically, please wait"
-                    else:
-                        msg = (
-                            "Centring in progress. Please save"
-                            + " the suggested centring or re-center"
-                        )
 
-                    log.warning(msg)
-                else:
-                    dm.start_centring_method(dm.MANUAL3CLICK_MODE)
-
-                logging.getLogger("user_level_log").info("Centring ...")
-                centring_result = async_result.get()
-                if centring_result["valid"]:
-                    logging.getLogger("user_level_log").info("Centring done !")
-                else:
-                    if centring_method == queue_entry.CENTRING_METHOD.FULLY_AUTOMATIC:
-                        raise queue_entry.QueueSkippEntryException(
-                            "Could not center sample, skipping", ""
+        use_custom_centring_routine = HWR.beamline.diffractometer.get_property(
```

Comment:
i like this... What are you using this for? (XRay centring comes to my mind)

Reply:
Well, its mostly used on A1 where we wrote some PMAC routine to speed up phase change and perform cantering on the MD computer. Its work mostly done by Didier Nurizzo. There are actually also some workarounds to issues of the MD firmware, so we can now specify weather to use those or not